blocks/generator.go

```diff
@@ -167,6 +167,12 @@ func (g *Generator) run() error {
 			if len(g.optimisticOutput) > 0 {
 				g.processOptimisticLayers(maxLayer)
 			}
+			if err := proposals.Delete(g.cdb, out.Layer); err != nil {
```

Comment:
what about old proposals and transactions link? will it conflict with sync if someone will request proposals a bit later?

i thought that it may be better to prune old proposals & certs every 30m or so in separate goroutine, and run  autovacuum there as well but every day instead

Reply:
i was thinking this can be shipped ASAP, as proposals is the major pain point and the db will stop growing at least.

later i will do something like you suggested and with vaccum because the timing to prune proposals and proposal_transactions are trickier.
one depends on hdist, and the other depends on mempool txs.

proposal syncing is only done during hare rounds. there it only prune up to layer N-1 at the end of hare for N.
so it's not likely that proposal sync will happen for old layers.

Reply:
btw, mempool never cross-referenced with the proposal tables so it's ok that proposals are deleted.